### ISSUE TITLE
Toggle Crossair freezes pymca

### ISSUE THREAD

**Issue by maurov** [vasole/pymca#840]
@vasole with Pymca 5.7.1 on the beamline, when opening Spec files and enabling the Crossair triggers the following error:

```
Traceback (most recent call last):
  File "<string>", line 525, in dispatcherAddSelectionSlot
  File "<string>", line 652, in _dispatcherAddSelectionSlot
  File "/users/blissadm/conda/miniconda/envs/sloth/lib/python3.8/site-packages/PyMca5/PyMcaGui/pymca/ScanWindow.py", line 454, in _addSelection
    self.resetZoom()
  File "/users/blissadm/conda/miniconda/envs/sloth/lib/python3.8/site-packages/PyMca5/PyMcaGraph/Plot.py", line 1097, in resetZoom
    self._plot.resetZoom(dataMargins)
  File "/users/blissadm/conda/miniconda/envs/sloth/lib/python3.8/site-packages/PyMca5/PyMcaGraph/backends/MatplotlibBackend.py", line 2354, in resetZoom
    self.graph.resetZoom(dataMargins)
  File "/users/blissadm/conda/miniconda/envs/sloth/lib/python3.8/site-packages/PyMca5/PyMcaGraph/backends/MatplotlibBackend.py", line 1458, in resetZoom
    xmin, xmax, ymin, ymax = self.getDataLimits('left')
  File "/users/blissadm/conda/miniconda/envs/sloth/lib/python3.8/site-packages/PyMca5/PyMcaGraph/backends/MatplotlibBackend.py", line 1518, in getDataLimits
    lineXMin = nanmin(x)
  File "/users/blissadm/conda/miniconda/envs/sloth/lib/python3.8/site-packages/PyMca5/PyMcaGraph/backends/MatplotlibBackend.py", line 56, in nanmin
    x = x[numpy.isfinite(x)]
TypeError: only integer scalar arrays can be converted to a scalar index
````

Any idea how to fix it?

Thanks,
Mauro

**Comment by vasole** [comment]
I guess with an explicit cast to integer. I take a look.

**Comment by vasole** [comment]
Please, can you tell me the version of numpy used?

**Comment by vasole** [comment]
I have patched your sloth environment with #841. It should be fine now.

**Comment by maurov** [comment]
> I have patched your sloth environment with #841. It should be fine now.

Great, thanks! I confirm it is fine now.
